tests/test_cli_files.py

```diff
@@ -241,3 +241,101 @@ def test_move_file_running_workflow():
+    response = {
+        "items": [
+            {
+                "last-modified": "2021-06-14T10:20:13",
+                "name": "data/names.txt",
+                "size": {"human_readable": "20 Bytes", "raw": 20},
+            },
+        ]
+    }
+    env = {"REANA_SERVER_URL": "localhost"}
+    mock_http_response, mock_response = Mock(), Mock()
+    mock_http_response.status_code = status_code
+    mock_response = response
+    reana_token = "000000"
+    runner = CliRunner(env=env)
+    with runner.isolation():
+        with patch(
+            "reana_client.api.client.current_rs_api_client",
+            make_mock_api_client("reana-server")(mock_response, mock_http_response),
+        ):
+            result = runner.invoke(
+                cli,
+                [
+                    "ls",
+                    "-t",
+                    reana_token,
+                    "--workflow",
+                    "mytest.1",
+                    "--filter",
+                    "name=names",
+                    "--json",
+                ],
+            )
+            json_response = json.loads(result.output)
+            assert result.exit_code == 0
+            assert isinstance(json_response, list)
+            assert len(json_response) == 1
+            assert "names" in json_response[0]["name"]
+
+
+def test_list_files_filter_with_filename():
+    """Test list workflow workspace files with filter and filename."""
+    status_code = 200
+    response = {
+        "items": [
+            {
+                "last-modified": "2021-06-14T10:20:14",
+                "name": "workflow/cwl/helloworld-slurmcern.cwl",
+                "size": {"human_readable": "965 Bytes", "raw": 965},
+            },
+            {
+                "last-modified": "2021-06-14T10:20:14",
+                "name": "workflow/cwl/helloworld-job.yml",
+                "size": {"human_readable": "122 Bytes", "raw": 122},
+            },
+            {
+                "last-modified": "2021-06-14T10:20:14",
+                "name": "workflow/cwl/helloworld.cwl",
+                "size": {"human_readable": "867 Bytes", "raw": 867},
+            },
+        ]
+    }
+    env = {"REANA_SERVER_URL": "localhost"}
+    mock_http_response, mock_response = Mock(), Mock()
+    mock_http_response.status_code = status_code
+    mock_response = response
+    reana_token = "000000"
+    runner = CliRunner(env=env)
+    with runner.isolation():
+        with patch(
+            "reana_client.api.client.current_rs_api_client",
+            make_mock_api_client("reana-server")(mock_response, mock_http_response),
+        ):
+            result = runner.invoke(
+                cli,
+                [
+                    "ls",
+                    "-t",
+                    reana_token,
+                    "--workflow",
+                    "mytest.1",
+                    "**/*.cwl",
+                    "--filter",
+                    "last-modified=2021-06-14",
```

Comment:
Ditto here, the matching should be substring, not exact match:

```console
$ reana-client ls -w helloworld-cwl-kubernetes --filter last-modified=2021-06-14T14:33:28
NAME            SIZE   LAST-MODIFIED      
workflow.json   1287   2021-06-14T14:33:28
inputs.json     145    2021-06-14T14:33:28

$ reana-client ls -w helloworld-cwl-kubernetes --filter last-modified=2021               
NAME   SIZE   LAST-MODIFIED

$ reana-client ls -w helloworld-cwl-kubernetes --filter last-modified=T   
NAME   SIZE   LAST-MODIFIED
```

Reply:
Same here
<img width="812" alt="Screenshot 2021-06-14 at 4 56 08 PM" src="https://user-images.githubusercontent.com/24358501/121913164-70883b80-cd31-11eb-955b-4fa76d8efc89.png">

Reply:
This works after r-w-controller PR update :+1: